cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -144,7 +145,8 @@ controller, and serviceaccounts controller.`,
 				os.Exit(1)
 			}
 
-			if err := Run(c.Complete(), wait.NeverStop); err != nil {
+			stopCh := server.SetupSignalHandler()
+			if err := Run(c.Complete(), stopCh); err != nil {
```

Comment:
why is this PR not in upstream?

Reply:
This is being worked on, unfortunately didn't land in 1.22. We're hoping to finish that in 1.23.